__tests__/app.test.js

```diff
@@ -171,3 +165,67 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("responds with 201 and add a new comment to the the comments with a username and body keys", () => {
+    return request(app)
+      .post(`/api/articles/4/comments`)
+      .send({ username: "butter_bridge", body: "I've been amazed!" })
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toMatchObject({
+          author: "butter_bridge",
+          body: "I've been amazed!",
+          article_id: 4,
+          votes: 0,
+          comment_id: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  test("responds with 400 when the article_id is not a valid number", () => {
+    return request(app)
+      .post("/api/articles/invalidID/comments")
+      .send({ username: "butter_bridge", body: "This is a comment" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid data type");
+      });
+  });
+  test("responds with 404 when the article_id is valid but does not exist", () => {
+    return request(app)
+      .post("/api/articles/999/comments")
+      .send({ username: "butter_bridge", body: "This is a comment" })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Article not found");
+      });
+  });
+  test("responds with 400 when the username or body is missing from the request body", () => {
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send({ username: "butter_bridge" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Missing required fields: username and body");
+      });
+  });
+  test("responds with 400 when the username or body is not of the correct data type", () => {
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send({ username: 123, body: "My comment here" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid data type for username or body");
+      });
+  });
+  test("responds with 404 when the username does not exist", () => {
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send({ username: "non_existent_user", body: "My comment here" })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("User not found");
+      });
+  });
```

Comment:
really nice thorough testing suite 👍 
You could consider adding one more test here. A status 201: ignores unnecessary properties